File: front/src/types/project.ts
import type { Task } from "./task";

export type Project = {
  id: string;
  title: string;
  description: string | null;
  createdAt: string;
  tasks?: Task[];
};

export type CreateProjectInput = {
  title: string;
  description?: string;
};

export type UpdateProjectInput = {
  title?: string;
  description?: string;
};

export type ProjectQuery = {
  page: number;
  size: number;
  title?: string;
  description?: string;
};

File: src/main/java/com/example/demo/domain/entity/Project.java
package com.example.demo.domain.entity;

import com.example.demo.application.dto.project.CreateProjectDTO;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.MappedCollection;
import org.springframework.data.relational.core.mapping.Table;

import java.sql.Timestamp;
import java.util.List;
import java.util.UUID;

@Data
@Table(name = "projects", schema = "test_nekit")
public class Project {

    @Id
    @Column("id")
    private UUID id;

    @Column("title")
    private String title;

    @Column("description")
    private String description;

    @Column("created_at")
    private Timestamp createdAt;

    @MappedCollection(idColumn = "project_id")
    private List<Task> tasks;

    public Project(CreateProjectDTO dto) {

        this.title = dto.title();
        this.description = dto.description();
        this.createdAt = new Timestamp(System.currentTimeMillis());
    }
}